app/jobs/resolution_proofing_job.rb

```diff
@@ -66,82 +68,27 @@ def make_vendor_proofing_requests(
     applicant_pii:,
     threatmetrix_session_id:,
     request_ip:,
-    should_proof_state_id:
+    should_proof_state_id:,
+    double_address_verification:
   )
-    device_profiling_result = proof_with_threatmetrix_if_needed(
+    result = resolution_proofer.proof(
       applicant_pii: applicant_pii,
-      user: user,
+      user_email: user&.confirmed_email_addresses&.first&.email,
       threatmetrix_session_id: threatmetrix_session_id,
       request_ip: request_ip,
+      should_proof_state_id: should_proof_state_id,
+      double_address_verification: double_address_verification,
       timer: timer,
     )
 
-    add_threatmetrix_proofing_component(user.id, device_profiling_result) if user.present?
-
-    resolution_result = timer.time('resolution') do
-      resolution_proofer.proof(applicant_pii)
-    end
-
-    state_id_result = Proofing::StateIdResult.new(
-      success: true, errors: {}, exception: nil, vendor_name: 'UnsupportedJurisdiction',
-    )
-    if should_proof_state_id && user_can_pass_after_state_id_check?(resolution_result)
-      timer.time('state_id') do
-        state_id_result = state_id_proofer.proof(applicant_pii)
-      end
-    end
-
-    result = Proofing::ResolutionResultAdjudicator.new(
-      resolution_result: resolution_result,
-      state_id_result: state_id_result,
-      should_proof_state_id: should_proof_state_id,
-      device_profiling_result: device_profiling_result,
-    ).adjudicated_result.to_h
+    log_threatmetrix_info(result.device_profiling_result, user)
+    add_threatmetrix_proofing_component(user.id, result.device_profiling_result) if user.present?
```

Comment:
> This change increases the probability that the job can stop executing between when we call ThreatMetrix and when we log and save the results.

I think you're referring specifically to ThreatMetrix results, correct?  If the job stops executing between when we call ThreatMetrix and when we log and save the Threatmetrix results, it would definitely make it harder to investigate ThreatMetrix API and integration issues.  If I move `log_threatmetrix_info` and `add_threatmetrix_proofing_component` into `progressive_proofer.rb`, would that address your concerns?

Also, could you say more about how this "change increases the probability that the job can stop executing between when we call ThreatMetrix and when we log and save the results," please?  I want to be sure I follow, thanks.

Reply:
> I think you're referring specifically to ThreatMetrix results, correct?

Yes, that's correct.

> If I move `log_threatmetrix_info` and `add_threatmetrix_proofing_component` into `progressive_proofer.rb`, would that address your concerns?

That could be used in one strategy for addressing my concern here. The important detail is to minimize processing and especially long-lived processes between when we call ThreatMetrix and when we log the results.

One more consideration here is that engineers who are viewing the logs will often interpret the logged events as taking place in the order they have been logged. They are also likely to interpret API calls as taking place close to the time(s) recorded in the log entry. Accommodation to both of these notions is improved by keeping the logging much closer to the API call in the code.

> Also, could you say more about how this "change increases the probability that the job can stop executing between when we call ThreatMetrix and when we log and save the results," please?

I'm specifically thinking here about situations in which the process could terminate - fatal errors like using too much memory, uncaught exceptions related to the API calls, VM shutdown, and similar. These issues aren't quite as visible in situations where the application infrastructure is matched well to the load, but they are inevitable in the long run. If these situations aren't handled well, then they can be quite hard to remediate after the fact.

Part of what I hoped would be evaluated in answering this question - and it may not be very consequential, not sure yet - is also the impact this kind of change might have if the system fails to save the ThreatMetrix result due to process termination. If we move the logging and saving back next to the API call, though, then I doubt this would increase any associated risk.

Reply:
Hi!  I looked into this more since we paired earlier today and I wanted to follow up with you.  I also reached out to Team Ada, in order to get context from developers who have been on the project for longer.  Here are my takeaways.
If the resolution proofing job fails:

* it is unlikely that logging the threatmetrix info immediately after the call to threatmetrix, rather than after the calls to lexisnexis and aamva, will greatly aid developers in the job failure investigation process
* we will have to re-run the job no matter what, so it is okay if the proofing component is not created before the job fails. It can be created when the job is run again


Additionally, moving the threatmetrix logs and threatmetrix proofing component creation to the progressive proofer gives the progressive proofer another concern.  For the above reasons, I decided to leave these two lines in `resolution_proofing_job`.